File: android/app/src/main/java/com/ets3d/bluebirdalertsandroid/Labels.kt
package com.ets3d.bluebirdalertsandroid

object AppLabels {
    const val KEY_LOCKDOWN = "lockdown"
    const val KEY_EVACUATION = "evacuation"
    const val KEY_SHELTER = "shelter"
    const val KEY_SECURE = "secure"
    const val KEY_REQUEST_HELP = "request_help"

    const val LOCKDOWN = "Lockdown"
    const val EVACUATION = "Evacuation"
    const val SHELTER = "Shelter"
    const val SECURE = "Secure Perimeter"
    const val REQUEST_HELP = "Request Help"

    const val ACTIVE_HELP_REQUESTS = "Active Help Requests"
    const val NO_ACTIVE_HELP_REQUESTS = "No active help requests."
    const val FORWARD_REQUEST_HELP = "Forward Request Help"

    val DEFAULT_FEATURE_LABELS: Map<String, String> = mapOf(
        KEY_LOCKDOWN to LOCKDOWN,
        KEY_EVACUATION to EVACUATION,
        KEY_SHELTER to SHELTER,
        KEY_SECURE to SECURE,
        KEY_REQUEST_HELP to REQUEST_HELP,
    )

    private fun normalizeFeatureKey(value: String): String {
        return when (value.trim().lowercase()) {
            "team_assist", "team assist", "request help" -> KEY_REQUEST_HELP
            else -> value.trim().lowercase()
        }
    }

    fun labelForFeatureKey(
        key: String,
        remoteLabels: Map<String, String> = DEFAULT_FEATURE_LABELS,
    ): String {
        val normalized = normalizeFeatureKey(key)
        return remoteLabels[normalized] ?: DEFAULT_FEATURE_LABELS[normalized] ?: key
    }

    fun featureDisplayName(
        rawValue: String,
        remoteLabels: Map<String, String> = DEFAULT_FEATURE_LABELS,
    ): String {
        return labelForFeatureKey(rawValue, remoteLabels)
    }
}

File: BlueBirdAlerts/BlueBirdAlerts/Labels.swift
import Foundation

enum AppLabels {
    static let keyLockdown = "lockdown"
    static let keyEvacuation = "evacuation"
    static let keyShelter = "shelter"
    static let keySecure = "secure"
    static let keyRequestHelp = "request_help"

    static let lockdown = "Lockdown"
    static let evacuation = "Evacuation"
    static let shelter = "Shelter"
    static let secure = "Secure Perimeter"
    static let requestHelp = "Request Help"

    static let activeHelpRequests = "Active Help Requests"
    static let noActiveHelpRequests = "No active help requests."
    static let forwardRequestHelp = "Forward Request Help"

    static let defaultFeatureLabels: [String: String] = [
        keyLockdown: lockdown,
        keyEvacuation: evacuation,
        keyShelter: shelter,
        keySecure: secure,
        keyRequestHelp: requestHelp,
    ]

    static func normalizeFeatureKey(_ value: String) -> String {
        let normalized = value.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
        switch normalized {
        case "team_assist", "team assist", "request help":
            return keyRequestHelp
        default:
            return normalized
        }
    }

    static func labelForFeatureKey(_ key: String, overrides: [String: String] = defaultFeatureLabels) -> String {
        let normalized = normalizeFeatureKey(key)
        return overrides[normalized] ?? defaultFeatureLabels[normalized] ?? key
    }

    static func featureDisplayName(for rawValue: String, overrides: [String: String] = defaultFeatureLabels) -> String {
        let normalized = normalizeFeatureKey(rawValue)
        switch normalized {
        case keyLockdown, keyEvacuation, keyShelter, keySecure, keyRequestHelp:
            return labelForFeatureKey(normalized, overrides: overrides)
        default:
            return rawValue
        }
    }
}
